### PR TITLE
NUnit3 support: download and finding console exe.

### PR DESCRIPTION
**Pull request by f00f**
It's just a first step, command line syntax seems to be different as well.

### CODE REVIEW

File: src/main/groovy/com/ullink/gradle/nunit/NUnit.groovy [hunk 1]
```diff
@@ -39,7 +39,9 @@ class NUnit extends ConventionTask {
 
     File getNunitExec() {
         assert getNunitHome(), "You must install NUnit and set nunit.home property or NUNIT_HOME env variable"
-        File nunitExec = nunitBinFile("nunit-console${useX86 ? '-x86' : ''}.exe")
```
Comment (gluck) [inline src/main/groovy/com/ullink/gradle/nunit/NUnit.groovy:42]: x86 seems to be a console param instead on NUnit v3 (as they're using a separate daemon to actually run the tests)

**Comment by f00f** [comment]
- Added support for renamed command line parameters.
- Fixed a bug which caused a crash and the test faiures.

Some parameters still don't work (e.g. -exclude, -include) and I'm not sure what to do about -noshadow. However, I don't use these, yet, so I'll leave this task to someone else.

File: src/main/groovy/com/ullink/gradle/nunit/NUnit.groovy [hunk 2]
```diff
@@ -39,7 +41,9 @@ class NUnit extends ConventionTask {
...
+        File nunitExec = getNunitVersion().startsWith("3.")
```
Comment (florian-gandon) [inline src/main/groovy/com/ullink/gradle/nunit/NUnit.groovy]: What about a property "isV3 in the task ?

File: src/main/groovy/com/ullink/gradle/nunit/NUnit.groovy [hunk 3]
```diff
@@ -39,7 +41,9 @@ class NUnit extends ConventionTask {
...
+            ? nunitBinFile('nunit3-console.exe')
+            : nunitBinFile("nunit-console${useX86 ? '-x86' : ''}.exe")
```
Comment (reeflog) [inline src/main/groovy/com/ullink/gradle/nunit/NUnit.groovy]: I would log a warning about the usage of useX86 when it's V3 as it's ignored.

File: src/main/groovy/com/ullink/gradle/nunit/NUnitPlugin.groovy [hunk 4]
```diff
@@ -52,7 +52,12 @@ class NUnitPlugin implements Plugin<Project> {
         if (!nunitCacheDir.exists()) {
             nunitCacheDir.mkdirs()
         }
-        def ret = new File(nunitCacheDir, NUnitName)
+        def nunitCacheDirForVersion = new File(nunitCacheDir, NUnitName)
+
+        // special handling for nunit3 flat zip file
+        def zipOutputDir = version.startsWith("3.") ? nunitCacheDirForVersion : nunitCacheDir;
```
Comment (florian-gandon) [inline src/main/groovy/com/ullink/gradle/nunit/NUnitPlugin.groovy]: The download is working ? The repo on github doesn't have the same name (nunit intead of nunitv2).

File: src/main/groovy/com/ullink/gradle/nunit/NUnit.groovy (hunk 2 above)
Comment (f00f) [inline src/main/groovy/com/ullink/gradle/nunit/NUnit.groovy]: Yeah, I like how you solved that.
I had a similar thing, but got confused with all the implicitness of gradle/Groovy and finally decided to remove it again.
I might port your approach.

File: src/main/groovy/com/ullink/gradle/nunit/NUnitPlugin.groovy (hunk 4 above)
Comment (f00f) [inline src/main/groovy/com/ullink/gradle/nunit/NUnitPlugin.groovy]: Well, for sure I had to change the download URL, but I cannot find that change in the commits now. Strange...

File: src/main/groovy/com/ullink/gradle/nunit/NUnit.groovy (hunk 1 above)
Comment (f00f) [inline src/main/groovy/com/ullink/gradle/nunit/NUnit.groovy:42]: Added in 01d515ab21b32de26e586ae05641d3f592950f69

File: src/main/groovy/com/ullink/gradle/nunit/NUnit.groovy (hunk 2 above)
Comment (f00f) [inline src/main/groovy/com/ullink/gradle/nunit/NUnit.groovy]: Merged in bd44ad5625c0c7a3377e1311b698222a0d9296e9

File: src/main/groovy/com/ullink/gradle/nunit/NUnitPlugin.groovy (hunk 4 above)
Comment (f00f) [inline src/main/groovy/com/ullink/gradle/nunit/NUnitPlugin.groovy]: Fixed in bd44ad5625c0c7a3377e1311b698222a0d9296e9

**Comment by florian-gandon** [comment]
LGTM

**Pull request merged**